### DISCUSSION TITLE
chap 5 질문 정리

### DISCUSSION THREAD

**Discussion by inseo24** [JS-OS-study/hyokyung-and-students#15] [General]
chap 5 질문 정리 discussion입니다.

**Comment by inseo24** [comment]
<h3>느낀 점 & 궁금한 점 </h3>

- 정보처리기사 공부하면서 대충 알았던 CPU 스케줄링에 대해서 배워서 재밌었다. 생각보다 내용이 어렵지 않아서 그 때 더 열심히 공부할 걸 하는 생각이 들었다.
- multilevel processor 스케줄링부터 설명을 들으면서도 이해가 안됐던 거 같다. 
- Homogeneous 프로세서는 어떻게 알아서 각 프로세서가 꺼내가는지,,, 피피티에 나온 내용처럼 특정 job이 특정 프로세서에서 실행되어야 할 경우 스케줄링이 어떻게 될지 등 단순히 이런게 있구나하는 식으로 넘긴 거 같다.

<hr />
<h3>문제</h3>

Q. CPU 스케줄러가 필요한 이유가 뭘까요?
A. ________ job과 _______ job이 섞여 있기 때문이다.

**Reply by 1chz** [reply]
`cpu bound job`과 `i/o bouind job`이 섞여 있기 때문이다.

**Reply by pullingoff** [reply]
IO bound, CPU bound

**Reply by sihyung92** [reply]
IO bound , CPU bound

**Comment by sihyung92** [comment]
# 느낀 점 & 궁금한 점
- 스케줄링의 각종 알고리즘에 대해 학습하고, 효과에 대해 이해할 수 있어서 좋았다. 단순 암기를 넘어 싱글코어에서 각 알고리즘이 필요한 이유, 대화형 프로그래밍에서 어떤 식으로 스케줄링을 해결하고 있는지 알게 되어 좋았다. 
- 수업에는 간단하게 나오지만 공룡책에는 실시간 CPU 스케줄링에서 soft 실시간시스템과 hard 실시간 시스템에 대한 상세한 알고리즘들이 나오는데 (rate-monotonic 스케줄링, earliest deadline first 스케줄링, POSIX 실시간 스케줄링 )이해하기 어려워 아쉬웠다.

# 문제
Q. 다음 중 싱글코어 스케줄링 알고리즘이 아닌 것을 고르시오.
1. 선입선처리(FCFS)
2. Symmetric Multiprocessing (SMP)
3. Round Robin (RR)
4. Shortest Job First (SJF)

**Reply by pullingoff** [reply]
스케줄링 방법이 진짜 다양하군여...!!!

**Reply by 1chz** [reply]
2

**Reply by pullingoff** [reply]
2번 SMP

**Reply by inseo24** [reply]
2번 SMP

**Comment by 1chz** [comment]
# 느낀점

---

- CPU 스케쥴링 알고리즘 개발에 실패한 경우들을 보니 알고리즘 개발자들도 사람이라는 걸 느끼게 됐다.
- CPU 스케쥴링 알고리즘들을 보니 의외로 다른 시스템들에서 CPU 스케쥴링 알고리즘들을 많이 채용하여 사용하고 있다고 느꼈다.

# 문제

---

- _________는 대부분의 현재 대부분의 운영체제가 채택하고 있는 CPU 스케쥴링 기법으로, 쉽게 이야기 해보자면 ___________ 알고리즘에 ______을 구현한 것이다.

**Reply by pullingoff** [reply]
라운드 로빈 스케쥴링 기법 / 우선순위 스케쥴링 + Aging 기법

**Reply by sihyung92** [reply]
Round Robin, FCFS, time slice

**Reply by inseo24** [reply]
라운드로빈, FCFS, 선점형?

**Reply by 1chz** [reply]
MLFQ / MLQ / Aging

**Comment by pullingoff** [comment]
# 느낀 점
1. 라운드 로빈이 time quantum을 어떻게 정하느냐에 큰 영향을 받는다는게 인상깊었다. 이 시간 할당량이 계속 바뀌는건지, 아니면 한번 설정되면 그대로 가는건지 찾아봐야겠다. (어떤 기준으로 설정하는 지도 궁금)
# 질문
1. Starvation을 해결하는 방법으로는 _A_ 기법이 있다. _A_ 기법을 사용하면 우선순위가 낮은 프로세스더라도 오래 기다리면 우선순위를 높여준다.
2. 멀티 프로세서 스케쥴링에서 일부 프로세서에 잡이 몰리지 않도록 부하를 적절하게 공유하는 것을 영어로 ______라고 한다.

**Reply by 1chz** [reply]
1. Aging
2. Load sharing or Load balancing

**Reply by sihyung92** [reply]
Aging, load sharing

**Reply by inseo24** [reply]
1. Aging
2. load balancing
